portafly/src/i18n/i18n.tsx

```diff
@@ -0,0 +1,34 @@
+import i18n, { FormatFunction } from 'i18next'
+import LanguageDetector from 'i18next-browser-languagedetector'
+import { initReactI18next } from 'react-i18next'
+import { ITranslationsPages, Translations, EN } from 'i18n'
+
+const formatFn: FormatFunction = (value, format) => {
+  if (format === 'uppercase') return value.toUpperCase()
+  if (format === 'lowercase') return value.toLowerCase()
+  return value
+}
```

Comment:
This fn makes me thing a lot if you consider it's type `FormatFunction`... value could be `any` and format any `string` or `undefined`... It's sad, because it's enforced by I18Next... you could craft something more robust and elegant like:

```ts
type Format = 'uppercase' | 'lowercase'

const formatFn = (value: string, format?: Format): string => (
  format ? {
    uppercase: value.toUpperCase(),
    lowercase: value.toLowerCase()
  }[format] : value
)
```

Reply:
I'm sure we can somehow enforce our own type for `FormatFunction`. Maybe thru our `useTranslation` wrapper hook.